Mage.Sets/src/mage/cards/a/ArcadeGannon.java

```diff
@@ -0,0 +1,179 @@
+        this.addAbility(
+                new SimpleStaticAbility(new ArcadeGannonEffect(filter))
+                        .setIdentifier(MageIdentifier.OnceEachTurnCastWatcher)
+                        .addHint(OnceEachTurnCastWatcher.getHint()),
+                new OnceEachTurnCastWatcher()
+        );
+    }
+
+    private ArcadeGannon(final ArcadeGannon card) {
+        super(card);
+    }
+
+    @Override
+    public ArcadeGannon copy() {
+        return new ArcadeGannon(this);
+    }
+}
+
+class ArcadeGannonEffect extends AsThoughEffectImpl {
+
+    private final FilterCard filter;
+
+    ArcadeGannonEffect(FilterCard filter) {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.WhileOnBattlefield, Outcome.Benefit);
+        this.filter = filter;
+        this.staticText = "<i>For Auld Lang Syne </i>Once during each of your turns, you may cast " + filter.getMessage() + " from your graveyard with mana value less than or equal to the number of quest counters on Arcade Gannon.";
+    }
+
+    private ArcadeGannonEffect(final ArcadeGannonEffect effect) {
+        super(effect);
+        this.filter = effect.filter;
+    }
+
+    @Override
+    public ArcadeGannonEffect copy() {
+        return new ArcadeGannonEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        throw new IllegalArgumentException("Wrong code usage: can't call applies method on empty affectedAbility");
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability affectedAbility, Ability source, Game game, UUID playerId) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent sourcePermanent = source.getSourcePermanentIfItStillExists(game);
+        if (controller == null || sourcePermanent == null) {
+            return false;
+        }
+        if (game.isActivePlayer(playerId) // only during your turn
+                && source.isControlledBy(playerId) // only you may cast
+                && Zone.GRAVEYARD.equals(game.getState().getZone(objectId)) // from graveyard
+                && affectedAbility instanceof SpellAbility // characteristics to check
+        ) {
+            SpellAbility spellAbility = (SpellAbility) affectedAbility;
+            Card cardToCheck = spellAbility.getCharacteristics(game);
+            if (spellAbility.getManaCosts().isEmpty() || spellAbility.getCharacteristics(game).getManaValue() > game.getPermanent(source.getSourceId()).getCounters(game).getCount(CounterType.QUEST)) {
+                return false;
+            }
+            return spellAbility.spellCanBeActivatedRegularlyNow(playerId, game)
+                    && filter.match(cardToCheck, playerId, source, game);
+        }
+        return false;
+    }
+}
+
+class CastFromGraveyardOnceWatcher extends Watcher {
```

Comment:
Why are you duplicating an existing watcher with the same name? Can't it just be used as is?